kyo-prelude/shared/src/main/scala/kyo/kernel/Effect.scala

```diff
@@ -42,4 +61,10 @@ object Effect:
                 f(ex)
         end try
     end catching
+
+    private[kyo] def defer[A, S](f: Safepoint ?=> A < S)(using _frame: Frame): A < S =
```

Comment:
I decided to make this method `private[kyo]` because I can't see a use case that would make sense for the user call it directly. Stack safety is provided automatically and side effects should be suspended via `IO`.